config/routes.rb

```diff
@@ -13,6 +13,12 @@
   get 'user/info' => 'users#info'
   get 'user/authenticate' => 'users#authenticate'
 
+  get 'document/v2/open' => 'documents_v2#open'
+  post 'document/v2/save' => 'documents_v2#save'
```

Comment:
The save should be a put instead of a post.  

I also wish that we could use more REST like URLs. So it would be be like: `v2/document/:key` and then the action will be determined by the http verb: GET -> open, PUT -> save, post/patch -> patch.  The problem is that the current code is relying on knowing if it is a readOnly or readWriteKey.  If all keys were unique then this could work because we could just look for the key in both fields.  Or there could be a `readOnly=true` param passed.

The next problem is how to implement `copy_shared` in this RESTful approach. The typical Rails thing we've done is `v2/documents/:key/copy_shared`.  That doesn't seem great, but it is simplier than other options I've seen.